src/coreclr/vm/jitinterface.h

```diff
@@ -565,18 +565,218 @@ class CEEInfo : public ICorJitInfo
+    void NibbleMapSet();
+
+    virtual void getEHinfo(
+                            CORINFO_METHOD_HANDLE ftn,              /* IN  */
+                            unsigned      EHnumber,                 /* IN */
+                            CORINFO_EH_CLAUSE* clause               /* OUT */
+                          ) override final;
+
+    void setEHinfoWorker(
+                          EE_ILEXCEPTION* pEHInfo,
+                          unsigned      EHnumber,
+                          const CORINFO_EH_CLAUSE* clause
+                        );
+
+    EECodeGenManager*       m_jitManager;   // responsible for allocating memory
+    void*                   m_CodeHeader;   // descriptor for JITTED code - read/execute address
+    void*                   m_CodeHeaderRW; // descriptor for JITTED code - code write scratch buffer address
+    size_t                  m_codeWriteBufferSize;
+    BYTE*                   m_pRealCodeHeader;
+    HeapList*               m_pCodeHeap;
+    COR_ILMETHOD_DECODER *  m_ILHeader;     // the code header as exist in the file
+
+#if defined(_DEBUG)
+    ULONG                   m_codeSize;     // Code size requested via allocMem
+#endif
+
+    size_t                  m_GCinfo_len;   // Cached copy of GCinfo_len so we can backout in BackoutJitData()
+    size_t                  m_EHinfo_len;   // Cached copy of EHinfo_len so we can backout in BackoutJitData()
+
+    ULONG32                 m_iOffsetMapping;
+    ICorDebugInfo::OffsetMapping * m_pOffsetMapping;
+
+    ULONG32                 m_iNativeVarInfo;
+    ICorDebugInfo::NativeVarInfo * m_pNativeVarInfo;
+
+    ICorDebugInfo::InlineTreeNode    *m_inlineTreeNodes;
+    ULONG32                           m_numInlineTreeNodes;
+    ICorDebugInfo::RichOffsetMapping *m_richOffsetMappings;
+    ULONG32                           m_numRichOffsetMappings;
+
+    // The first time a call is made to CEEJitInfo::GetProfilingHandle() from this thread
+    // for this method, these values are filled in.   Thereafter, these values are used
+    // in lieu of calling into the base CEEInfo::GetProfilingHandle() again.  This protects the
+    // profiler from duplicate calls to its FunctionIDMapper() callback.
+    struct GetProfilingHandleCache
+    {
+        GetProfilingHandleCache() :
+            m_bGphIsCacheValid(false),
+            m_bGphHookFunction(false),
+            m_pvGphProfilerHandle(NULL)
+        {
+            LIMITED_METHOD_CONTRACT;
+        }
+
+        bool                    m_bGphIsCacheValid : 1;        // Tells us whether below values are valid
+        bool                    m_bGphHookFunction : 1;
+        void*                   m_pvGphProfilerHandle;
+    } m_gphCache;
+};
 
 // CEEJitInfo is the concrete implementation of callbacks that the EE must provide for the JIT to do its
 // work.   See code:ICorJitInfo#JitToEEInterface for more on this interface.
-class CEEJitInfo : public CEEInfo
+class CEEJitInfo : public CEECodeGenInfo
```

Comment:
Nit, I think you can use `class CEEJitInfo final : public CEECodeGenInfo` instead of scattering final throughout the definition of `CEEJitInfo`. Also the same for `CInterpreterJitInfo`

Reply:
Right, a bulk of it was just a move / copy of the function headers from the original `CEEJitInfo` that had it per function. I will change it to per class except for the few ones that are final in the `CEECodeGenInfo` shared base class.

Reply:
Done